src/SmartTransactionsController.ts

```diff
@@ -566,36 +671,55 @@ export default class SmartTransactionsController extends BaseController<
       }
       const { nonceDetails } = nonceLock;
 
-      this.updateSmartTransaction({
-        chainId,
-        nonceDetails,
-        preTxBalance,
-        status: SmartTransactionStatuses.PENDING,
-        time,
-        txParams,
-        uuid: data.uuid,
-        cancellable: true,
-      });
+      this.#updateSmartTransaction(
+        {
+          nonceDetails,
+          preTxBalance,
+          status: SmartTransactionStatuses.PENDING,
+          time,
+          txParams,
+          uuid: data.uuid,
+          cancellable: true,
+        },
+        { chainId, ethersProvider },
+      );
     } finally {
       nonceLock.releaseLock();
     }
 
     return data;
   }
 
+  getChainId(networkClientId?: NetworkClientId): Hex {
+    if (!networkClientId) {
+      return this.config.chainId;
+    }
+    return this.getNetworkClientById(networkClientId).configuration.chainId;
+  }
+
+  getProvider(networkClientId?: NetworkClientId): any {
+    if (!networkClientId) {
+      return this.ethersProvider;
+    }
+    return this.getNetworkClientById(networkClientId).provider;
+  }
+
   // TODO: This should return if the cancellation was on chain or not (for nonce management)
   // After this successful call client must update nonce representative
   // in transaction controller external transactions list
-  async cancelSmartTransaction(uuid: string): Promise<void> {
-    const { chainId } = this.config;
+  async cancelSmartTransaction(
+    uuid: string,
+    networkClientId?: NetworkClientId,
```

Comment:
this interface is odd. I would expect chainId to come off the transaction object itself. Not asking for any change, just pointing this out

Reply:
In this case there is no transaction object? Just a uuid